pkg/apis/troubleshoot/v1beta2/collector_shared.go

```diff
@@ -144,6 +144,7 @@ type CopyFromHost struct {
 	Timeout         string            `json:"timeout,omitempty" yaml:"timeout,omitempty"`
 	HostPath        string            `json:"hostPath" yaml:"hostPath"`
 	ExtractArchive  bool              `json:"extractArchive,omitempty" yaml:"extractArchive,omitempty"`
+	RetryLimit      int32             `json:"retryLimit,omitempty" yaml:"retryLimit,omitempty"`
```

Comment:
This is not what I had in mind. My intention was to gracefully exit if the collectd hostpath does not exist. We would not retry since we assume the path will never appear. This would be our opinionated way of saying `collectd` is not installed. We do not want to give users more knobs to adjust (`retryLimit` extra parameter). Users who choose to collect these same files but would like to continuously retry until the hostpath is available, or timeout can use the [copy from host](https://troubleshoot.sh/docs/collect/copy-from-host/) collector.

Think of this extra bit as an implementation detail of the `collectd` collector. We need it cause the code is shared by both collectors.